src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F

```diff
@@ -609,8 +607,9 @@ subroutine sfclayer_from_MPAS(configs,mesh,diag_physics,sfc_input,its,ite)
        v10_sea(i,j)     = 0._RKIND
 
        !overwrite some local variables for sea-ice cells:
-       if(xice_p(i,j).ge.xice_threshold .and. xice_p(i,j).le.1._RKIND) then
+       if(xice_p(i,j).ge.xice_threshold .and. xice_p(i,j).lt.1._RKIND) then
           xland_sea(i,j)  = 2._RKIND
+          xland_p(i,j)    = 1._RKIND
```

Comment:
is this line necessary? are there any consequences to this line when sfc_update is activated? in my understanding, `xland` is changed from water (2) to non-water (1) or vice versa in two places during a simulation, at initialization when seaice is present and when reading an xice field from update_surface. This would add a third location and I'm trying to think of a scenario where this is necessary.

Reply:
Previously, xland_p and xland_sea are both set to be equal to xland. When the ice is fractional we have to make sure that xland_p=1 (ice) and xland_sea=2 (water). There is a possible scenario when they both are 2 on the previous time step, and at the current time step the sea ice became higher than the xice_threshold value, so the xland_p should be set to 1.

Reply:
I guess I don't see how sea ice is changing between time steps if it is not read in from the surface update file, and in that module xland is being set to 1 for fractional sea ice points already. I did not think that sea ice was predicted in the model.

So, it seems that was there previously is still correct. The only thing that needs to be changed is xland_sea being set to 2 and that is already there.

Reply:
Some advanced sea ice models include melting/growing of sea ice. So, the scenario when xland_p=2 is possible.